### PR TITLE
Add sample for multi-project jacoco reports using tasks

### PR DESCRIPTION
**Pull request by bmuskalla**

### CODE REVIEW

**Comment by bmuskalla** [comment]
@jjohannes Can you please take another look before we proceed? (only the groovy sample)

File: subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/README.adoc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+With the plugin applied, it automatically attaches itself to the `test` task to collect the code coverage.
+Please refer to the link:{userManualPath}/jacoco_plugin.html[JaCoCo chapter] for more details on the plugin configuration.
+
+To generate a report spanning all our tested projects, we introduce a new task on the root project (or an aggregator subproject if you like) that takes care of setting up the reports, where jacoco can find the source files and which projects to collect the coverage data from.
+
+====
+include::sample[dir="groovy",files="build.gradle[tags=coverageTask]"]
+include::sample[dir="kotlin",files="build.gradle.kts[tags=coverageTask]"]
+====
+
```
Comment (ljacomet) [inline subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/README.adoc:25]: I would add a mention here about the lazy APIs we leverage to make sure there is no ordering issue.

File: subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/kotlin/build.gradle.kts [hunk 2]
```diff
@@ -6,3 +11,41 @@ allprojects {
...
+
+                // NOTE: the report task does *not* depend on the `test` task by default. Meaning you have to ensure
+                // that `test` (or other tasks generating code coverage) run before generating the report.
+                // You can achieve this by calling the `check` lifecycle task manually
+                // $ ./gradlew test codeCoverageReport
+
+                // Alternatively you can declare a dependency so that the `codeCoverageReport` depends on the `test` tasks.
+                // This inevitably enforces that running the `codeCoverageTask` runs the test tasks as well.
+                // This may be your intended behavior.
+                rootProject.tasks.findByName("codeCoverageReport")!!.dependsOn(testTask)
```
Comment (jjohannes) [inline subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/kotlin/build.gradle.kts:35]: :o: Avoid `!!` - use `tasks .getByName()` or even better `tasks["codeCoverageReport"]`

**Comment by bmuskalla** [comment]
Need to check the dependency example again, due to the new approach, I think the task dependencies are setup too late now

**Comment by bmuskalla** [comment]
@bot-gradle test QF

File: subprojects/docs/docs.gradle [hunk 3]
```diff
@@ -287,8 +287,6 @@ samples {
                 from(templates.javaModuleInfoForListLibrary)
                 from(templates.javaModuleInfoForUtilitiesLibrary)
                 from(templates.javaModuleInfoForApplication)
-                from(templates.javaJunit5TestForApplication)
-                from(templates.javaJunit5TestForListLibrary)
```
Comment (jjohannes) [inline subprojects/docs/docs.gradle:291]: :x: These should not be removed

File: subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/groovy/build.gradle [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+tasks.register("codeCoverageReport", JacocoReport) {
+    // If a subproject applies the 'jacoco' plugin, add the result it to the report
+    subprojects { subproject ->
+        subproject.plugins.withType(JacocoPlugin).configureEach {
+            subproject.tasks.matching({ t -> t.extensions.findByType(JacocoTaskExtension) }).configureEach { testTask ->
+                sourceSets subproject.sourceSets.main
+                executionData(testTask)
+            }
+
+            // Alternatively you can declare a dependency so that the `codeCoverageReport` depends on the `test` tasks.
```
Comment (jjohannes) [inline subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/groovy/build.gradle:36]: :o: It's probably not clear that "Alternatively" here now relates to the text block further up. Instead you could give the full context again, like:

> To automatically run `test` overtime `./gradlew codeCoverageReport` is called...

And the you can remove this sentence below:
"This inevitably enforces that running the `codeCoverageTask` runs the test tasks as well."

File: subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/kotlin/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    subprojects {
+        val subproject = this
+        subproject.plugins.withType<JacocoPlugin>().configureEach {
+            subproject.tasks.matching({ it.extensions.findByType<JacocoTaskExtension>() != null }).configureEach {
+                val testTask = this
+                sourceSets(subproject.sourceSets.main.get())
+                executionData(testTask)
+            }
+
+            // Alternatively you can declare a dependency so that the `codeCoverageReport` depends on the `test` tasks.
```
Comment (jjohannes) [inline subprojects/docs/src/samples/jvm-multi-project-with-code-coverage/kotlin/build.gradle.kts:38]: Same as in Groovy version...

File: subprojects/docs/docs.gradle (hunk 3 above)
Comment (bmuskalla) [reply subprojects/docs/docs.gradle:291]: Sorry, I meant to double check the merge conflict and forgot :| Thanks for catching

**Pull request merged**